apps/backend/src/lambdas/report.ts

```diff
@@ -1,16 +1,268 @@
+    (itm: PaymentEntity) => itm.status === MatchStatus.MATCH
+  );
+
+  const paymentExceptions = matchedPaymentsAndExceptions.filter(
+    (itm: PaymentEntity) => itm.status === MatchStatus.EXCEPTION
+  );
+
+  const pendingAndInProgressDeposits = await posDepositService.findPosDeposits(
+    { minDate: event.period.from, maxDate: event.period.to },
+    event.program,
+    locations.flatMap((itm: NormalizedLocation) => itm.merchant_ids),
+    [MatchStatus.PENDING, MatchStatus.IN_PROGRESS]
+  );
+  console.log(pendingAndInProgressDeposits, 'PENDING AND IN PROGRESS');
+  const matchedDepositsAndExceptions = await posDepositService.findPosDeposits(
+    { minDate: event.period.to, maxDate: event.period.to },
+    event.program,
+    locations.flatMap((itm: NormalizedLocation) => itm.merchant_ids),
+    [MatchStatus.MATCH, MatchStatus.EXCEPTION]
+  );
+  const matchedDeposits = matchedDepositsAndExceptions.filter(
+    (itm: POSDepositEntity) => itm.status === MatchStatus.MATCH
+  );
+  const depositExceptions = matchedDepositsAndExceptions.filter(
+    (itm: POSDepositEntity) => itm.status === MatchStatus.EXCEPTION
+  );
+  const matchedDepositIdsFromPayments =
+    matchedPayments.map((itm: PaymentEntity) => itm.pos_deposit_match) ?? [];
+  /*eslint-disable */
+  const matchedDepositsFromAnotherDay: POSDepositEntity[] = [];
+
+  matchedDepositIdsFromPayments.forEach((itm: POSDepositEntity | undefined) => {
+    if (itm) {
+      if (
+        !matchedDeposits.find((deposit: POSDepositEntity) => deposit === itm)
+      ) {
+        matchedDepositsFromAnotherDay.push(itm);
+      }
+    }
+  });
+  const depositsWithoutMatchedPayment = matchedDeposits.filter(
+    (itm: POSDepositEntity) =>
+      matchedPayments.find(
+        (pymnt: PaymentEntity) => pymnt.pos_deposit_match === itm
+      )
+  );
+  const matchedPaymentsFromAnotherDay =
+    await paymentService.findPaymentsByPosDeposits(
+      depositsWithoutMatchedPayment
+    );
+  const posPaymentsForReport = [
+    ...pendingAndInProgressPayments,
+    ...matchedPayments,
+    ...matchedPaymentsFromAnotherDay,
+    ...paymentExceptions,
+  ];
+  const posDepositsFroReport = [
+    ...pendingAndInProgressDeposits,
+    ...matchedDeposits,
+    ...matchedDepositsFromAnotherDay,
+    ...depositExceptions,
+  ];
+
+  return {
+    posPayments: posPaymentsForReport,
+    posDeposits: posDepositsFroReport,
+  };
+};
+
+const casReportData = async (
+  app: INestApplicationContext,
+  event: ReportConfig,
+  locations: NormalizedLocation[]
+) => {
+  const cashDepositService = app.get(CashDepositService);
+  const posDepositService = app.get(PosDepositService);
+  const maxDate = parse(event.period.to, 'yyyy-MM-dd', new Date());
+  /* extract the month from the "to-date"*/
+  const minDate = new Date(getYear(maxDate), getMonth(maxDate), 1);
+
+  const dateRange = {
+    minDate: format(minDate, 'yyyy-MM-dd'),
+    maxDate: format(maxDate, 'yyyy-MM-dd'),
```

Comment:
Please ensure that this isn't going back a day because of UTC

Reply:
logged: 

`{ minDate: '2023-05-01', maxDate: '2023-05-25' } CAS DATE RANGE`

when generating a report for May 25th